content/guides/docker-concepts/running-containers/multi-container-applications.md

```diff
@@ -0,0 +1,203 @@
+
+In this hands-on guide, you'll first see how to build and run a counter web application based on Node.js, Nginx reverse proxy and Redis database using the `docker run` commands. You’ll also see how you can simplify the entire deployment process using Docker Compose.
+
+## Running with the Docker CLI
+
+1. Get the sample application. If you have Git, you can clone the repository for the sample application. Otherwise, you can download the sample application. Choose one of the following options.
+
+   {{< tabs >}}
+   {{< tab name="Clone with git" >}}
+
+   Use the following command in a terminal to clone the sample application repository.
+
+   ```console
+   $ git clone https://github.com/dockersamples/nginx-node-redis
+   ```
+
+   {{< /tab >}}
+   {{< tab name="Download" >}}
+
+   Download the source and extract it.
+
+   {{< button url="https://github.com/dockersamples/nginx-node-redis/archive/refs/heads/main.zip" text="Download the source" >}}
+
+   {{< /tab >}}
+   {{< /tabs >}}
+
+2. Navigate into the `nginx-node-redis-main` directory
+
+    ```console
+    $ cd nginx-node-redis-main
+    ```
+
+    Inside this directory, you'll find two sub-directories - `nginx` and `web`.
+
+3. [Download and install](https://www.docker.com/products/docker-desktop/) Docker Desktop.
+
+### Build the images
+
+
+1. Navigate into the `nginx` directory to build the image by running the following command:
+
+
+    ```console
+    $ docker build -t nginx .
+    ```
+
+2. Navigate into the `web` directory and run the following command to build the first web image:
+    
+    ```console
+    $ docker build -t web .
+    ```
+
+### Run the containers
+
+1. Before you can run a multi-container application, you need to create a network for them all to communicate through. You can do so using the `docker network create` command:
+
+    ```console
+    $ docker network create sample-app
+    ```
+
+2. Start the Redis container by running the following command, which will attach it to the previously created network and create a network alias (useful for DNS lookups):
+
+    ```console
+    $ docker run -d  --name redis --network sample-app --network-alias redis redis
+    ```
+
+3. Start the first web container by running the following command:
+
+    ```console
+    $ docker run -d --name web1 -h web1 --network sample-app --network-alias web1 web
+    ```
+
+4. Start the second web container by running the following:
+
+    ```console
+    $ docker run -d --name web2 -h web2 --network sample-app --network-alias web2 web
+    ```
+    
+5. Start the Nginx container by running the following command:
+
+    ```console
+    $ docker run -d --name nginx --network sample-app  -p 80:80 nginx
+    ```
+
+     > Nginx is typically used as a reverse proxy for web applications, routing traffic to backend servers. In our case, it routes to the Node.js backend containers (web1 or web2)
+
+
+6.  Verify the containers are up by running the following command:
+
+    ```console
```

Comment:
Suggest separating input and output into different codeblocks because it makes it easier to use the copy button.

Reply:
Fixed.